components/bl-innerImageZoom-component/src/index.js

```diff
@@ -0,0 +1,47 @@
+import { useMemo, useState } from 'react';
+
+const { cn } = BackendlessUI.CSSUtils;
+const DEFAULT_ZOOM_POSITION = '0% 0%';
+
+export default function InnerImageZoomComponent({ component, elRef, eventHandlers }) {
+  const { classList, style, display, source, zoomIn } = component;
+  const { onMouseOver, onMouseOut, onMouseMove } = eventHandlers;
+
+  const zoom = Math.max(100, zoomIn);
+  const [zoomPosition, setZoomPosition] = useState(DEFAULT_ZOOM_POSITION);
+  const figureStyle = useStyles(source, zoomPosition, zoom);
```

Comment:
There are no strict rules, but try to split it in some order.

For example here we could

```
const { classList, style, display, source, zoomIn } = component; //Works with props
const { onMouseOver, onMouseOut, onMouseMove } = eventHandlers;

const [zoomPosition, setZoomPosition] = useState(DEFAULT_ZOOM_POSITION);//Default hooks like useState useRef useMemo

const figureStyle = useStyles(source, zoomPosition, zoom); // Our custom hooks

const zoom = Math.max(100, zoomIn); //regular calculations
```